File: uikit/src/main/java/com/sendbird/uikit/tasks/TaskQueue.java
package com.sendbird.uikit.tasks;

import androidx.annotation.NonNull;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;

public class TaskQueue {
    @NonNull
    final private static ExecutorService taskExecutor = Executors.newCachedThreadPool();

    @NonNull
    public static <T> Future<T> addTask(@NonNull JobTask<T> task) {
        return taskExecutor.submit(task.getCallable());
    }

    @NonNull
    public static <T> Future<T> addTask(@NonNull JobResultTask<T> task) {
        return taskExecutor.submit(task.getCallable());
    }
}

File: uikit/src/main/java/com/sendbird/uikit/internal/tasks/TaskQueue.kt
package com.sendbird.uikit.internal.tasks

import java.util.concurrent.Executors
import java.util.concurrent.Future

internal object TaskQueue {
    private val taskExecutor = Executors.newCachedThreadPool()

    @JvmStatic
    fun <T> addTask(task: JobTask<T>): Future<T> {
        return taskExecutor.submit(task.callable)
    }

    @JvmStatic
    fun <T> addTask(task: JobResultTask<T>): Future<T> {
        return taskExecutor.submit(task.callable)
    }
}
